source/installation/restore-minio.rst

```diff
@@ -0,0 +1,265 @@
+
+.. _minio-restore-hardware-failure:
+
+==============================
+Recover after Hardware Failure
+==============================
+
+.. default-domain:: minio
+
+.. contents:: Table of Contents
+   :local:
+   :depth: 1
+
+Distributed MinIO deployments rely on :ref:`Erasure Coding
+<minio-erasure-coding>` to provides built-in tolerance for multiple disk or node
+failures. Depending on the deployment topology, MinIO can tolerate the loss of
+up to half the drives or nodes in the deployment while maintaining read access
+to stored objects.
+
+While MinIO can operate in a degraded state without significant performance
+loss, administrators should replace failed hardware at the earliest convenience
+to mitigate the risk of more significant failures.
+
+The following table lists the typical types of failure in a MinIO deployment
+and links to procedures for recovering from each:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 30 70
+   :width: 100%
+
+   * - Failure Type
+     - Description
+
+   * - :ref:`Drive Failure <minio-restore-hardware-failure-drive>`
+     - MinIO supports hot-swapping failed drives with new healthy drives. 
+
+   * - :ref:`Node Failure <minio-restore-hardware-failure-node>`
+     - MinIO detects when a node rejoins the deployment and begins aggressively
+       healing data previously stored on that node.
+
+.. admonition:: MinIO Professional Support
+   :class: note
+
+   |subnet| provides 24/7 direct-to-engineering support for MinIO 
+   deployments. Consider leveraging SUBNET to ensure smooth and reliable
+   operation of MinIO deployments, including root-cause analysis, 
+   health diagnostics, and architecture reviews.
+
+   You can alternatively seek support on the 
+   `MinIO Community Slack <https://minio.slack.com>`__. Community Support
+   is best-effort only and has no SLAs around responsiveness.
+
+.. _minio-restore-hardware-failure-drive:
+
+Drive Failure Recovery
+----------------------
+
+MinIO supports hot-swapping failed drives with new healthy drives. MinIO
+detects and aggressively heals those drives without performing any restart
+procedure.
+
+The procedure can be summarized as follows:
+
+1. Unmount the failed drive.
+2. Replace the failed drive with a new drive of the same type and capacity.
+3. Format the drive to match the filesystem (e.g. xfs) and set a matching label.
+4. Update ``/etc/fstab`` (if necessary) and remount the drive.
+5. Monitor MinIO logs to verify detection and healing of the drive.
+6. Use :mc-cmd:`mc admin heal` to inspect healing status.
+
+The replacement drive hardware should be substantially similar to the failed
+drive. For example, replace a failed SSD with another SSD drive of the same
+capacity. While you can use drives with larger capacity, MinIO uses the
+*smallest* drive's capacity as the ceiling for all drives in the :ref:`Server
+Pool <minio-intro-server-pool>`.
+
```

Comment:
Do not attempt to manually recover data from the old drive, even if some of it can be recovered. MinIO will start its own healing which will ensure that data is consistent after healing and should just be given a newly formatted disk for this.

Reply:
Are there any files we would want a user or customer to keep for later diagnostics? e.g .minio.sys or something? 

I vaguely recall we have used that file in the past to do certain kinds of debugging.

Reply:
Nah. They don't have to reformat the old disk :D

But they definitely shouldn't begin manually recovering data.

Reply:
Change addressed above.